plugins/casutg/src/main/java/com/blazemeter/jmeter/threads/concurrency/ConcurrencyThreadStarter.java

```diff
@@ -53,14 +70,32 @@ private long getPlannedConcurrency() {
         }
 
         if (rampUp == 0 || timeOffset > rampUp) {
-            return Math.round(maxConcurr);
+            return maxConcurr;
         } else if (steps > 0) {
             double stepSize = maxConcurr / (double) steps;
             double stepLen = rampUp / (double) steps;
             return Math.round(stepSize * (Math.floor(timeOffset / stepLen) + 1));
         } else {
-            double slope = maxConcurr / rampUp;
+            double slope = (double) maxConcurr / rampUp;
             return Math.round(slope * timeOffset);
         }
     }
+
+    void checkNeedsPropertiesReloading(long now) {
```

Comment:
Maybe make it private and call in constructor to avoid code duplication?

Reply:
As you guessed later for JUNIT